src/react-app-env.d.ts

```diff
@@ -1 +1,9 @@
 /// <reference types="react-scripts" />
+interface Window {
+  ethereum?: {
+    isMetaMask?: true
+    on?: (...args: any[]) => void
+    removeListener?: (...args: any[]) => void
+    autoRefreshOnNetworkChange?: boolean
+  }
+}
```

Comment:
I am not sure but does this extend the `window` interface or overwrite it? It is the latter then we should make sure this extends.

Reply:
It does extend the interface I think but let me confirm